memory_backend.go

```diff
@@ -0,0 +1,286 @@
+package neoq
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/guregu/null"
+	"github.com/iancoleman/strcase"
+	"github.com/jsuar/go-cron-descriptor/pkg/crondescriptor"
+	"github.com/pkg/errors"
+	"github.com/robfig/cron"
+	"golang.org/x/exp/slog"
+)
+
+const (
+	DefaultQueueCapacity = 10000 // the default capacity of individual queues
+	EmptyCapacity        = 0
+)
+
+// MemBackend is a memory-backed neoq backend
+type MemBackend struct {
+	config       *memConfig
+	cron         *cron.Cron
+	logger       Logger
+	mu           *sync.Mutex // mutext to protect mutating state on a pgWorker
+	jobCount     int64       // number of jobs that have been queued since start
+	handlers     sync.Map    // map queue names [string] to queue handlers [Handler]
+	fingerprints sync.Map    // map fingerprints [string] to their jobs [Job]
+	futureJobs   sync.Map    // map jobIDs [int64] to [Jobs]
+	queues       sync.Map    // map queue names [string] to queue handler channels [chan Job]
+}
+
+type memConfig struct{}
```

Comment:
Future proofing?

Reply:
Yeah I thought there would be a use for this pretty quickly, but it never turned up. Since its internal API, I'm going to remove it now.